src/components/AndelaCentres/AndelaCentresComponent.jsx

```diff
@@ -0,0 +1,135 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { isEmpty } from 'lodash';
+import {
+  Header,
+  Divider,
+  Table,
+  Segment,
+  Pagination
+} from 'semantic-ui-react';
+
+import TableRow from '../TableRowComponent';
+import NavBarComponent from '../../_components/NavBarContainer';
+import LoaderComponent from '../../components/LoaderComponent';
+import ItemsNotFoundComponent from '../common/ItemsNotFoundComponent';
+import DropdownComponent from '../../_components/DropdownComponent';
+
+import rowOptions from '../../_utils/pageRowOptions';
+
+class AndelaCentresComponent extends React.Component {
+  state = {
+    limit: 10,
+    activePage: 1
+  };
+
+  componentDidMount() {
+    this.props.loadAndelaCentres(this.state.activePage);
+  }
+
+  handleRowChange = (e, data) => {
+    this.setState({ limit: data.value });
+    this.props.loadAndelaCentres(this.state.activePage, data.value);
+  };
+
+  handlePaginationChange = (e, { activePage }) => {
+    this.setState({ activePage });
+    this.props.loadAndelaCentres(activePage);
+  };
+
+  getTotalPages = () => Math.ceil(this.props.centresCount / this.state.limit);
+
+  render() {
+    if (this.props.isLoading) {
+      return (
+        <NavBarComponent>
+          <LoaderComponent />
+        </NavBarComponent>
+      );
+    }
+
+    if (!this.props.isLoading && isEmpty(this.props.centres)) {
```

Comment:
This block can be combined with the last return block. See the implementation in [UsersComponent](https://github.com/AndelaOSP/art-dashboard/blob/develop/src/components/User/UsersComponent.jsx#L126)